From 9b4c60a89c2a55f36454b950a86246b6b86a9681 Mon Sep 17 00:00:00 2001
From: KonstantIMP <mihedovkos@gmail.com>
Date: Sat, 8 May 2021 22:14:57 +0300
Subject: [PATCH 1/2] Add linux 5.12.x support Kernel with version >= 5.12.x
 doesnt support GRO_DROP so the driver must be updated

---
 os_dep/linux/recv_linux.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/os_dep/linux/recv_linux.c b/os_dep/linux/recv_linux.c
index 7fecc843..fc06a030 100644
--- a/os_dep/linux/recv_linux.c
+++ b/os_dep/linux/recv_linux.c
@@ -359,7 +359,8 @@ static int napi_recv(_adapter *padapter, int budget)
 			if (rtw_napi_gro_receive(&padapter->napi, pskb) != GRO_DROP)
 				rx_ok = _TRUE;
 #else
-			rx_ok = _TRUE;
+			rtw_napi_gro_receive(&padapter->napi, pskb);
+            rx_ok = _TRUE;
 #endif
 			goto next;
 		}

From 1dfc05ad530e61d9f6f7d48c8ab85ac3ab6b4f85 Mon Sep 17 00:00:00 2001
From: KonstantIMP <43513955+KonstantIMP@users.noreply.github.com>
Date: Sat, 8 May 2021 22:18:14 +0300
Subject: [PATCH 2/2] Add linux 5.12.x support

Kernel with version >= 5.12.x doesnt support GRO_DROP so the driver must be updated
---
 os_dep/linux/recv_linux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_dep/linux/recv_linux.c b/os_dep/linux/recv_linux.c
index fc06a030..86a6d7d9 100644
--- a/os_dep/linux/recv_linux.c
+++ b/os_dep/linux/recv_linux.c
@@ -360,7 +360,7 @@ static int napi_recv(_adapter *padapter, int budget)
 				rx_ok = _TRUE;
 #else
 			rtw_napi_gro_receive(&padapter->napi, pskb);
-            rx_ok = _TRUE;
+                        rx_ok = _TRUE;
 #endif
 			goto next;
 		}
